packages/graphql/src/translate/translate-resolve-reference.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Node } from "../classes";
+import createProjectionAndParams from "./create-projection-and-params";
+import type { GraphQLOptionsArg, Context } from "../types";
+import { createAuthPredicates } from "./create-auth-and-params";
+import { AUTH_FORBIDDEN_ERROR } from "../constants";
+import { createMatchClause } from "./translate-top-level-match";
+import Cypher from "@neo4j/cypher-builder";
+
+export function translateResolveReference(
```

Comment:
I see a lot of copy/pasted code here, would it be possible to move part of it to a separate function?

Reply:
I've trimmed it down a lot now, I don't think this is necessary any more.